src/components/box/box.component.tsx

```diff
@@ -33,18 +31,24 @@ export interface BoxProps
   extends SpaceProps,
     LayoutProps,
     FlexboxProps,
-    ColorProps,
+    TagProps,
     Omit<PositionProps, "zIndex"> {
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   as?: keyof JSX.IntrinsicElements | React.ComponentType<any>;
```

Comment:
**issue** (kind-of): So as we're not spreading ...rest onto `<StyledBox>` and we've moved the `calculatePosition` function into `box.style.ts` it might be possible that `as` is not being passed or applied, potentially. I've got quite a bit to take a look at so can't do too much in depth testing but it may be worth taking a look